projects/plugins/jetpack/extensions/blocks/contact-form/transforms.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { pickBy, identity } from 'lodash';
+import { createBlock } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+const getAttributeValue = ( tag, att, content ) => {
+	const dbqts = content.match( new RegExp( `\\[${tag}[^\\]]* ${att}="([^"]*)"`, 'im' ) );
+    if ( dbqts != null && dbqts.length > 0 ) {
+        return dbqts[ 1 ];
+    }
+
+	const sinqts = content.match( new RegExp( `\\[${tag}[^\\]]* ${att}='([^']*)'`, 'im' ) );
+	if ( sinqts != null && sinqts.length > 0 ) {
+		return sinqts[ 1 ];
+    }
+
+	const noqts = content.match( new RegExp( `\\[${tag}[^\\]]* ${att}=([^\\s]*)\\s`, 'im' ) );
+    if ( noqts != null && noqts.length > 0 ) {
+       return noqts[ 1 ];
+    }
+
+	return false;
+};
+
+const getContactFieldBlockName = ( type ) => {
+    const prefix = 'jetpack';
+
```

Comment:
just a personal preference thing, but with this many switch options I find creating a lookup structure a slightly more concise, eg.:
```
const fieldTypes = {
	text: `${ prefix }/field-text`,
	url: `${ prefix }/field-text`,
	textarea: `${ prefix }/field-textarea`,
	radio: `${ prefix }/field-radio`,
	checkbox: `${ prefix }/field-checkbox`,
	select: `${ prefix }/field-select`,
	email: `${ prefix }/field-email`,
	name: `${ prefix }/field-name`,
	default: `${ prefix }/field-text`,
};
return fieldTypes[type] ? fieldTypes[type] : fieldTypes.default;
```